src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/ExpressionLambdaRewriter.cs

```diff
@@ -303,7 +303,7 @@ private BoundExpression Indices(ImmutableArray<BoundExpression> expressions)
             foreach (var arg in expressions)
             {
                 var index = Visit(arg);
-                if (arg.Type != _int32Type)
+                if (index.Type != _int32Type)
```

Comment:
Consider removing the `if` conditions instead, especially since the expressions were always true previously.